go/libraries/doltcore/env/repo_state.go

```diff
@@ -169,3 +174,44 @@ func (rs *RepoState) WorkingHash() hash.Hash {
 func (rs *RepoState) StagedHash() hash.Hash {
 	return hash.Parse(rs.Staged)
 }
+
+func (rs *RepoState) IsMergeActive() bool {
+	return rs.Merge != nil
+}
+
+func (rs *RepoState) GetMergeCommit() string {
+	return rs.Merge.Commit
+}
+
+// TODO: This functionality is currently unsupported by the SQL session.
+func (rs *RepoState) GetAllValidDocDetails() ([]doltdb.DocDetails, error) {
```

Comment:
I'm actually not sure now that you mention it...

Reply:
Ok I think I get it now. Since the sqle is taking in a repo state reader derived from dEnv it able to deal with doc changes. That might have to change in the future.